### PR TITLE
Prototype

### PR DESCRIPTION
**Pull request by jonfur**
Ready for PR of this functionality. Add text to input field, click button to have reversed string printed.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        <script>window.jQuery || document.write('<script src="js/vendor/jquery-{{JQUERY_VERSION}}.min.js"><\/script>')</script>
+        <script src="js/plugins.js"></script>
+        <script src="js/main.js"></script>
+
+        <!-- Google Analytics: change UA-XXXXX-Y to be your site's ID. -->
+        <script>
+            window.ga=function(){ga.q.push(arguments)};ga.q=[];ga.l=+new Date;
+            ga('create','UA-XXXXX-Y','auto');ga('send','pageview')
+        </script>
+        <script src="https://www.google-analytics.com/analytics.js" async defer></script>
```
Comment (whilestevego) [inline index.html]: A lot of this boiler plate is of no consequence to your project. I would remove some of it if I were you. Specifically, I'd remove lines 35-40, 9-12 and 19-21.

File: js/main.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+//Helper functions
+function $q (cssQuery) {
```
Comment (whilestevego) [inline js/main.js]: I realize I wrote `cssQuery` for the argument name, but that's what came to mind at the time. I think that `selector` is argument name that much better represents what you expect.

File: js/main.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+};
+function $el (element) {
+  return document.createElement(element);
+};
+
+//Cached Nodes
+var theField = $q("#the-field");
+var theH1 = $q("#the-real-h1");
+var body = document.body;
+var theButn = $q("#the-butn");
```
Comment (whilestevego) [inline js/main.js]: I highly encourage to come up with better the names than the ones I chose hastily. Something more closely represents the elements purpose which is not always easy, but that's one of the great difficulties of programming. :sweat_smile:

File: js/main.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+//String reversal function
+function reversy (s) {
```
Comment (whilestevego) [inline js/main.js]: A better name for this function might be `reverse`.

**Comment by whilestevego** [comment]
Aside from changing a few of the names, everything looks good and works! Ping me once you've implemented my comments! Cheers.

**Comment by jonfur** [comment]
Implemented your suggestions!

**Comment by whilestevego** [comment]
It looks like you accidentally committed that work you thought you lost. You'll have to remove it from this PR as it is unrelated to its scope. I recommend that you return to this PR's branch and run the following command:

```
git reset @~1
```

That will reset the code in the last commit where `@` refers to the top commit and `~1` decrements in history.  The commit will disappear from history and its code will be unstaged. It'll give you chance to recommit it with only what you want.

To push it up again, you'll have to force it. I trust find out how that is done.

**Pull request closed without merging**